Step2D[] footsteps_2d